OpenRA.Mods.Common/Traits/Buildings/ProductionAirdrop.cs

```diff
@@ -112,11 +103,142 @@ public override bool Produce(Actor self, ActorInfo producee, string productionTy
 					Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech", info.ReadyAudio, self.Owner.Faction.InternalName);
 				}));
 
-				actor.QueueActivity(new FlyOffMap(actor, Target.FromCell(w, endPos)));
-				actor.QueueActivity(new RemoveSelf());
+				actor.QueueActivity(new CallFunc(() =>
+				{
+					var exitPosition = deliveryPath.CalculateExitPosition(actor);
+					actor.QueueActivity(new FlyOffMap(actor, Target.FromCell(w, exitPosition)));
```

Comment:
Since `exitPosition` is always defined, there seems to be no need for `FlyOffMap` here. We can just queue a regular `Fly`.